### ISSUE TITLE
Image *actually* doesn't fill sprite horizontally for repeat-x background

### ISSUE THREAD

**Issue by kidkuro** [Compass/compass#999]
Issue #362 reported that images were not stretching to horizontally fill the sprite when given a `repeat` of `repeat-x` in a [`sprite-map()`](http://compass-style.org/reference/compass/helpers/sprites/#sprite-map)  invocation.

In that case, it turned out that the issue reporter had mistakenly used `$new-repeat: repeat-x` in the invocation instead of the name of the image, i.e. instead of the correct `$my-bg-repeat: repeat-x`, and there was actually nothing wrong.

The issue as it was described is now _actually_ happening, even with the correct syntax.

We have been successfully using `sprite-map()` with `repeat-x` to generate sprites that correctly stretch the image horizontally to fill the sprite, much like as was achieved using vanilla spriting in [this blog post](http://www.pfvdw.com/blog/compass-sprites-and-background-repeat-x), but after upgrading to 0.12.2 of Compass, the image is no longer stretching.

I have verified that it was working correctly in 0.11.7 and that it does not work in 0.12.1.

We have a folder called `nav` containing various images including a narrow image `folder.png`. The code we are using is:

```
$nav-sprite: sprite-map("nav/*.png", $folder-repeat: repeat-x);
.bar {
    background: sprite($nav-sprite, folder) repeat-x;
}
```

This worked perfectly in 0.11.x but in 0.12.x the generated sprite image does not stretch the folder.png image horizontally.

The vanilla spriting approach still works, so while we would prefer to use the `sprite-map()` helper, we can work around it using the following code:

```
$nav-folder-repeat: repeat-x !default;
@import "nav/*.png";
@include all-nav-sprites;
.bar {
    @include nav-sprite(folder);
    background-repeat: repeat-x;
}
```

**Comment by kappernator** [comment]
I can confirm that this is still an issue with compass 0.12.2.
Is there any solution on the way?

**Comment by scottdavis** [comment]
Have you tried the latest alpha version?

**Comment by kappernator** [comment]
I tried it just now, and couldnt get it to work in Compass 0.13.alpha.4.

Oh btw, I am a Windows 7 user

**Comment by scottdavis** [comment]
So you want the `background-repeat` property to be `repeat-x` but, you do not want the sprite tiled down the x-axis in the map? I think I'm missing something.

**Comment by kidkuro** [comment]
No, the opposite. The problem is that when `repeat-x` is enabled, the image _isn't_ tiled along the x-axis, when it _should be_.

Using the vanilla spriting approach as described at the bottom of the ticket produces the correct results, the image is tiled horizontally to fill the width of the sprite image. The bug is that the tiling doesn't occur when using `sprite-map`.

**Comment by hakonkrogh** [comment]
I can confirm that this is still an issue. Using compass 12.2 on a w7 machine:

<code>
$icons: sprite-map("icons/*.png",
 $layout: smart,
 $spacing: 3px,
 $slider-pager-bg-repeat: repeat-x
);
</code>

The vanilla workaround works, but I am forced to use sprite-map as I am relying on the map in order to use "sprite-position" and "sprite-path"
